File: lang/codes/GameOm/OmFrontEnd/src/api/zone.js
import request from '@/utils/request'


// 区服列表
export function ZoneList() {
    return request({
        url: "/zone/api/zone",
        method: "get"
    })
}

// 添加区服
export function AddZone(data) {
    return request({
        url: "/zone/api/zone",
        method: "post",
        data: data,
        timeout: 600000
    })
}

// 区服操作
export function ManZone(data) {
    return request({
        url: "/zone/api/zone",
        method: "put",
        data: data,
        timeout: 600000
    })
}

File: lang/codes/om-game/backend/app/route/zone.py
from starlette.routing import Route
from starlette.requests import Request
from starlette.endpoints import HTTPEndpoint
from internal.utils.response import Response
from internal.utils.token import TokenHandle
from internal.utils.ansible import ZoneHandle


class ZoneEndpoint(HTTPEndpoint):
    async def get(self, request: Request) -> Response:
        pass

    async def post(self, request: Request) -> Response:
        token = request.headers.get("token")
        if not token:
            return Response(40000, msg="token is None")
        zone_info = await request.json()
        ZoneHandle(zone=zone_info).ansible_cmd
        return Response(20000, data=zone_info)

    async def put(self, request: Request) -> Response:
        pass


zone_route = [
    Route("/api/zone", ZoneEndpoint)
]